go/common/enclave.go

```diff
@@ -103,6 +103,9 @@ type Enclave interface {
 
 	// GetLogs returns all the logs matching the filter.
 	GetLogs(encryptedParams EncryptedParamsGetLogs) (EncryptedResponseGetLogs, error)
+
+	// HealthCheck returns whether the enclave is in a healthy state
+	HealthCheck() (bool, error)
```

Comment:
Should we just extend the `Status` method for this?

Reply:
Good q. I guess the status can be a subset of the health check. Like the HealthCheck can provide that info. 
But extending feels like it would be mixing the logic of whats healthy and at what stage are things.

Reply:
Ok